tests/multipleBackend/routes/routeBackbeat.js

```diff
@@ -752,5 +759,261 @@ describeSkipIfAWS('backbeat routes', () => {
                 },
             ], done);
         });
+
+        it('should not put delete tags if the source is not Azure and ' +
+        'if-unmodified-since header is not provided', done => {
+            const awsKey = uuid();
+            async.series([
+                next =>
+                    awsClient.putObject({
+                        Bucket: awsBucket,
+                        Key: awsKey,
+                    }, next),
+                next =>
+                    makeRequest({
+                        authCredentials: backbeatAuthCredentials,
+                        hostname: ipAddress,
+                        port: 8000,
+                        method: 'POST',
+                        path: '/_/backbeat/batchdelete',
+                        headers: {
+                            'x-scal-storage-class': awsLocation,
+                            'x-scal-tags': JSON.stringify({
+                                'scal-delete-marker': 'true',
+                                'scal-delete-service': 'lifecycle-transition',
+                            }),
+                        },
+                        requestBody: JSON.stringify({
+                            Locations: [{
+                                key: awsKey,
+                                dataStoreName: awsLocation,
+                            }],
+                        }),
+                        jsonResponse: true,
+                    }, next),
+                next =>
+                    awsClient.getObjectTagging({
+                        Bucket: awsBucket,
+                        Key: awsKey,
+                    }, (err, data) => {
+                        assert.ifError(err);
+                        assert.deepStrictEqual(data.TagSet, []);
+                        next();
+                    }),
+            ], done);
+        });
+
+        it('should not put tags if the source is not Azure and ' +
```

Comment:
I'm guessing the 'not Azure' is part because we don't support versioning on Azure, right? If that's the case, it might be better to use something more generic for the tests names like "if the source doesn't support versioning" or sth like that. Cos I believe GCP doesn't support versioning yet either

Reply:
It's because Azure supports conditional deletes with the `DateUnModifiedSince` value. We've leveraged that when the data is transitioning from Azure, while we do our own conditional check on the other clouds by performing a HEAD request and comparing last modified dates.